test/src/scene/incidentAngleTest.cpp

```diff
@@ -0,0 +1,81 @@
+#include <helpers/geometryData.hpp>
+#include <helpers/lidarHelpers.hpp>
+#include <helpers/sceneHelpers.hpp>
+#include <helpers/commonHelpers.hpp>
+#include <helpers/textureHelpers.hpp>
+#include <helpers/testPointCloud.hpp>
+
+#include <RGLFields.hpp>
+
+/**
+ * This test instantiates a large thin wall (100x100x0.1) at Z=10 and shoots parallel rays from different sensor positions and orientations.
+ * The wall is expected to be hit by all rays, and the incident angle is expected to be combination of sensor and box orientations.
+ * Slight variations of the sensor and wall position are expected to have no effect on the incident angle.
+ */
+
+struct IncidentAngleTest : public RGLTestWithParam<std::tuple<Vec3f, Vec3f, Vec3f, Vec3f>>
+{
+	static const inline auto WALL_POS = Vec3f(0, 0, 10.0f);
+	static const inline auto WALL_DIMS = Vec3f(200, 200, 0.1);
+};
+
+// clang-format off
+INSTANTIATE_TEST_SUITE_P(Parametrized, IncidentAngleTest, testing::Combine(
+	/* SensorPos     */ testing::Values(Vec3f(0, 0, 0), Vec3f(1, 1, 1)),
+	/* WallPosOffset */ testing::Values(Vec3f(0, 0, 0), Vec3f(1, 1, 1)),
+	/* SensorRotDeg  */ testing::Values(Vec3f(5, 0, 0), Vec3f(0, -15, 0), Vec3f(-25, 25, 0)),
+	/* WallRotDeg    */ testing::Values(Vec3f(-10, 0, 0), Vec3f(0, 20, 0), Vec3f(30, -40, 0))
+));
+// clang-format on
+
+
+TEST_P(IncidentAngleTest, wall)
+{
+	// Parameters
+	auto&& [sensorPos, wallPosOffset, sensorRotDeg, wallRotDeg] = GetParam();
+	Vec3f sensorRotRad = (std::numbers::pi_v<float> / 180.0f) * sensorRotDeg;
+	Vec3f wallRotRad = wallRotDeg * Vec3f(std::numbers::pi_v<float> / 180.0f);
+
+	// Expectations
+	Vec3f expectedNormal = Mat3x4f::rotationRad(wallRotRad) * Vec3f{0, 0, -1};
+	Vec3f rayDir = Mat3x4f::rotationRad(sensorRotRad) * Vec3f{0, 0, 1};
```

Comment:
Those two lines are not very elegant. 
I get it, but those 1 and -1, are not very generic.

Reply:
Added a comment.